### PR TITLE
Add websocket updates for room list

### PR DESCRIPTION
**Pull request by izakfr**
Closes: #183

### CODE REVIEW

File: frontend/packages/web/src/components/features/Subscription/Subscription.tsx [hunk 1]
```diff
@@ -36,18 +52,34 @@ export default function Subscription() {
     });
   }, []);
 
+  const handleRoomListMessage = (message: any) => {
+    // TODO: implement a way to unsubscribe from backend
+    setRoomMap({ ...roomMap, ...(message.data.room_map as RoomMap) });
```
Comment (benshumaker) [inline frontend/packages/web/src/components/features/Subscription/Subscription.tsx:57]: Why is this message.data.room_map but the other handlers just access message.data?

File: frontend/packages/web/src/components/features/Subscription/Subscription.tsx [hunk 2]
```diff
@@ -36,18 +52,34 @@ export default function Subscription() {
...
       switch (messageState.channel) {
+        case 'room_list':
+          handleRoomListMessage(messageState);
+          break;
         case 'room':
-          setRoom({ ...room, ...(messageState.data as Room) });
-          if (messageState.data.game_id) {
-            setGameSub(messageState.data.game_id);
-          }
+          handleRoomMessage(messageState);
```
Comment (benshumaker) [inline frontend/packages/web/src/components/features/Subscription/Subscription.tsx:79]: Nice

File: frontend/packages/web/src/components/features/Subscription/Subscription.tsx [hunk 3]
```diff
@@ -1,7 +1,15 @@
 import React, { useEffect, useMemo, useState } from 'react';
 import { useRecoilState } from 'recoil';
-import { Game, Room } from '../../../models';
-import { roomSubState, gameSubState, roomState, gameState } from '../../../store/atoms';
+import merge from 'ts-deepmerge';
```
Comment (benshumaker) [inline frontend/packages/web/src/components/features/Subscription/Subscription.tsx:3]: Why did you add ts-deepmerge? I couldn't even find where you used merge...

File: frontend/packages/web/src/components/features/RoomList/RoomList.tsx [hunk 4]
```diff
@@ -22,23 +29,32 @@ export default function RoomList() {
...
   return (
     <RoomListContainer>
       <h1 style={{ color: '#ffffff' }}>Join a Room</h1>
       <ListContainer>
-        {rooms.map((r) => {
+        {Object.values(roomMap).map((r: Room) => {
+          if (Object.keys(r.players).length === 0) {
+            return;
+          }
+          if (r.players[player.id]) {
```
Comment (benshumaker) [inline frontend/packages/web/src/components/features/RoomList/RoomList.tsx:55]: Why do we need to check if they're already in the room? Wouldn't that imply the aren't on this screen?

File: frontend/packages/web/src/components/features/RoomList/RoomList.tsx [hunk 5]
```diff
@@ -22,23 +29,32 @@ export default function RoomList() {
     [context.roomHttpService, context.roomWebSocketService]
   );
 
+  useEffect(() => {
+    setRoomMapSub(true);
+    return () => {
```
Comment (benshumaker) [inline frontend/packages/web/src/components/features/RoomList/RoomList.tsx:34]: Does a useEffect return function call when the component un-renders?

**Review by benshumaker — CHANGES_REQUESTED** [review]
Mostly good. I have a few minor comments and questions

File: frontend/packages/web/src/components/features/Subscription/Subscription.tsx (hunk 1 above)
Comment (izakfr) [reply frontend/packages/web/src/components/features/Subscription/Subscription.tsx:57]: Discussed this offline.

**Review by benshumaker — APPROVED** [review]
Big Win

**Pull request closed without merging**